scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -226,62 +274,152 @@ private static void askInteractively(Console console, UserInfo userInfo, Action
+            Properties sourceLoginProps = new Properties();
+            if (sourceLoginFile != null) {
+                try (InputStreamReader stream = new InputStreamReader(new FileInputStream(sourceLoginFile), Charsets.UTF_8)) {
+                    sourceLoginProps.load(stream);
+                } catch (Exception e) {
+                    exitWithErrorMessage("could not read source login file : " + sourceLoginFile, null, e);
+                }
+            } else if (userInfo != null) {
+                if (userInfo.getPassword() == null) {
+                    // password can be null in case of account deletion
+                    sourceLoginProps.put(userInfo.getLogin(), "");
+                } else {
+                    sourceLoginProps.put(userInfo.getLogin(), userInfo.getPassword());
+                }
+            }
+
+            Multimap<String, String> sourceGroupsMap = null;
+
+            if (sourceGroupFile != null) {
+                sourceGroupsMap = loadGroups(sourceGroupFile);
+            } else {
+                sourceGroupsMap = TreeMultimap.create();
+                if (userInfo != null && !userInfo.getGroups().isEmpty()) {
+                    for (String group : userInfo.getGroups()) {
+                        sourceGroupsMap.put(userInfo.getLogin(), group);
                     }
-                    props.remove(userInfo.getLogin());
-                    groupsMap.removeAll(userInfo.getLogin());
-                    System.out.println("Deleted user " + userInfo.getLogin() + " in " + loginFilePath);
-                    break;
+                }
+            }
+            Collection<String> sourceLoginNames = sourceLoginProps.stringPropertyNames();
+            if (sourceLoginNames.isEmpty()) {
+                sourceLoginNames = sourceGroupsMap.keySet();
             }
 
-            storeLoginFile(loginFilePath, props);
+            boolean bulkMode = sourceLoginNames.size() > 1;
+
+            for (String user : sourceLoginNames) {
+                UserInfo sourceUserInfo = new UserInfo();
+                sourceUserInfo.setLogin(user);
+                sourceUserInfo.setPassword((String) sourceLoginProps.get(user));
+                if (sourceGroupsMap.containsKey(user)) {
+                    sourceUserInfo.setGroups(sourceGroupsMap.get(user));
+                }
+
+                switch (action) {
+                    case CREATE:
+                        createAccount(pubKey, sourceUserInfo, loginFilePath, groupFilePath, destinationLoginProps, destinationGroupsMap);
+                        break;
+                    case UPDATE:
+                        updateAccount(pubKey, sourceUserInfo, loginFilePath, destinationLoginProps, destinationGroupsMap, bulkMode);
+                        break;
+                    case DELETE:
+                        deleteAccount(sourceUserInfo, loginFilePath, groupFilePath, destinationLoginProps, destinationGroupsMap);
+                        break;
+                }
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "USER " 7 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1192)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal " does not exist in login file : " 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1192)